Remove retired courses and child courses from Course Admin Modal Same As dropdown

natalynyu opened the pull request:
This PR removes the retired and child courses from the Course Admin Modal's Same As dropdown. A new API method, `getActiveParentCourses`, was added to get all of the non-retired, non-child courses, and these courses are directly passed into the Course Admin Modal via a prop called `parentCourses`.

This PR also fixes tests that broke as a result of merging new changes from `develop`.

## Types of changes
<!--
  What types of changes does your code introduce? Put an `x` in all the boxes
  that apply:
-->
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality
      to change)

## Checklist:
<!--
  Go over all the following points, and put an `x` in all the boxes that apply.
-->
- [x] I have run `eslint` on the code
- [x] I have added JSDoc for all of my code (where applicable)
- [x] I have added tests to cover my changes.

## Priority:
- [x] Normal <!-- New piece of functionality -->
- [ ] High <!-- Critical bug requiring urgent review -->

## Related Issues:
<!--
  Add the issue number in below that this PR is intended to address.

  Also mention any issues that this PR is related to, and if possible,
  provide more information regarding the relationship to the issues in
  question (i.e does the PR fix the issue, is it designed to fix another bug
  that is similar to the issue etc.)
-->
Fixes #651

<!--
  Attribution:
  PR Template adapted from: https://github.com/h5bp/html5-boilerplate/blob/master/.github/PULL_REQUEST_TEMPLATE.md
-->

natalynyu committed:
Add course controller and service to get the active

This commit adds the controller and service method to get the non-retired courses that do not have a sameAs value. The reason we are looking for courses without a sameAs value is because those courses with a sameAs value are children, and child courses cannot be parents of courses.

natalynyu committed:
Linter issue - add missing curly brace

natalynyu committed:
Add API method to get the non-retired, non-child courses

natalynyu committed:
Add ActiveParentCourses DTO

The response that we get back from the API will be an array of course ids and respective catalog numbers.

natalynyu committed:
Add test data for ActiveParentCourses

natalynyu committed:
Add tests to check client API

natalynyu committed:
Add course service tests

These unit tests check that the courses are returned where there are records in the database and an empty array is returned when there aren't.

natalynyu committed:
Update the Course Admin to pass down the active parent courses

Instead of sending all of the current courses in the form of ManageCourseResponseDTO[], we are now passing down the active parent courses (ActiveParentCourses[]). This is then used to populate the Course Modal Same As dropdown options.

natalynyu committed:
Update existing tests to fix the data being passed to courses prop

The courses prop should accept an array of active parent courses instead of a course in the form of a ManageCourseResponseDTO.

natalynyu committed:
Fix courses prop value and remove irrelevant test

The test is now irrelevant, because the handling of whether the child course is no longer handled on the client side. This logic is now handled on the server side.

natalynyu committed:
Update the test data for active parent courses to include the other test courses

natalynyu committed:
Merge branch 'develop' into bugfix/651-remove-retired-sameAs

natalynyu committed:
Merge branch 'develop' into bugfix/651-remove-retired-sameAs

natalynyu committed:
Merge branch 'develop' into bugfix/651-remove-retired-sameAs

natalynyu committed:
Debug commit

Urjikoo commented on tests/integration/e2e/MultiYearPlan.test.tsx:
```diff
@@ -116,50 +117,78 @@ describe('End-to-end Multi Year Plan tests', function () {
     context('when a course is a child of another', function () {
-      it('should display the faculty of its parent course', async function () {
-        await renderResult.findByText(childCourse.title, { exact: false });
+      it.only('should display the faculty of its parent course', async function () {
```
The build is failing because of the applied exclusive test, so we can remove `.only` to fix the build.

natalynyu replied:
As discussed during the developer standup, this test was marked as `.only` for debugging purposes only. Thanks!

natalynyu committed:
Fix end to end multi year plan test

These updates make the finding of test data more strict by enforcing that the parent and child courses that are found occur in the test semester specified, and this make sure that the term pattern is TERM_PATTERN.BOTH.

natalynyu committed:
Merge branch 'develop' into bugfix/651-remove-retired-sameAs

natalynyu committed:
Update CourseAdmin to pass down allCourses and parentCourses

With the update from PR #661 (https://github.com/seas-computing/course-planner/pull/661), allCourses are needed in order to populate the Course Admin modal's list of associated sameAs courses. The parentCourses are calculated in the server and are needed in order to populate the sameAs dropdown.

natalynyu committed:
Fix existing tests to account for the new Course Admin props

We now have both allCourses and parentCourses as props, so the existing tests needed to be updated to reflect this.

natalynyu committed:
Improve CourseAdmin integration test

Instead of hard-coding a value for the title, we are looking for a course using .findOneOrFail that has a sameAs value of null. When the sameAs value is null, that means that the course does not have a parent.

natalynyu committed:
Fix Schedule end to end test

This test was originally sporadically failing. The test was improved by looking for a test course instance with a non-null sameAs value in order to find a parent-child course pair as opposed instead of taking a course and updating the sameAs value in the course admin modal. This takes out several unnecessary steps and cuts down on the amount of time the test needs to run. One important note is to make sure that the test course instance has an isSEAS value of IS_SEAS.Y. Otherwise, the course will not show in the schedule - this was a blocker that took a while to figure out.

Urjikoo reviewed (approved):
looks great! merge away
